### PR TITLE
Refactor to separate DOH<->UDP from VPN logic

### PR DESCRIPTION
**Pull request by bemasc**
This is a step toward enabling access to DOH from a UDP
server.

### CODE REVIEW

File: Android/app/src/main/java/app/intra/util/DnsUdpQuery.java [hunk 1]
```diff
@@ -28,4 +35,33 @@
   public short sourcePort;
   public short destPort;
   public long timestamp;
+
+  // Returns the question name, type, and id  present in |dnsPacket|, as a DnsUdpQuery object.
+  // Assumes the DNS packet has been validated.
+  public static DnsUdpQuery fromUdpBody(byte[] data) {
```
Comment (alalamav) [inline Android/app/src/main/java/app/intra/util/DnsUdpQuery.java]: s/data/dnsPacketData (more descriptive name, also match comment)

File: Android/app/src/main/java/app/intra/DnsResolverUdpToHttps.java [hunk 2]
```diff
@@ -17,260 +17,51 @@
...
 import app.intra.util.Ipv4Packet;
 import app.intra.util.Ipv6Packet;
-import app.intra.util.Names;
 import app.intra.util.UdpPacket;
 import okhttp3.Call;
 import okhttp3.Callback;
 import okhttp3.Response;
 
 // Reads DNS requests over UDP from |tunFd|, forwards them to Google Public DNS through HTTPS and
 // writes them back to the tun interface. Takes ownership of the file descriptor.
```
Comment (alalamav) [inline Android/app/src/main/java/app/intra/DnsResolverUdpToHttps.java]: Update documentation.

Comment (bemasc) [reply Android/app/src/main/java/app/intra/DnsResolverUdpToHttps.java]: Done

File: Android/app/src/main/java/app/intra/util/DnsUdpQuery.java (hunk 1 above)
Comment (bemasc) [reply Android/app/src/main/java/app/intra/util/DnsUdpQuery.java]: Done

**Pull request merged**